src/libraries/CalldataBytesLib.sol

```diff
@@ -2,12 +2,47 @@
 pragma solidity ^0.8.0;
 
 /// @title Library for Bytes Manipulation
-library BytesLib {
+library CalldataBytesLib {
     error SliceOutOfBounds();
 
     /// @notice equivalent to SliceOutOfBounds.selector
     bytes4 constant SLICE_ERROR_SELECTOR = 0x3b99b53d;
 
+    /// @notice Decode the `_arg`-th element in `_bytes` as a dynamic array
+    /// @dev The decoding of `length` and `offset` is universal,
+    /// whereas the type declaration of `res` instructs the compiler how to read it.
+    /// @param _bytes The input bytes string to slice
+    /// @param _arg The index of the argument to extract
+    /// @return length Length of the array
+    /// @return offset Pointer to the data part of the array
+    function toLengthOffset(bytes calldata _bytes, uint256 _arg)
```

Comment:
why do we need toLengthOffset and toBytes? Not sure where they are being used?

Reply:
removed